cpp/benchmarks/common/generate_input.cu

```diff
@@ -274,128 +371,127 @@ struct stored_as<T, std::enable_if_t<std::is_same_v<T, bool>>> {
+    thrust::gather(
+      thrust::device, sample_indices.begin(), sample_indices.end(), samples.begin(), data.begin());
+    thrust::gather(thrust::device,
+                   sample_indices.begin(),
+                   sample_indices.end(),
+                   samples_null_mask.begin(),
+                   null_mask.begin());
   }
 
-  // cudf expects the null mask buffer to be padded up to 64 bytes. so allocate the proper size and
-  // copy what we have.
-  rmm::device_buffer result_bitmask{cudf::bitmask_allocation_size_bytes(num_rows),
-                                    rmm::cuda_stream_default};
-  cudaMemcpyAsync(result_bitmask.data(),
-                  null_mask.data(),
-                  null_mask.size() * sizeof(cudf::bitmask_type),
-                  cudaMemcpyHostToDevice,
-                  rmm::cuda_stream_default);
+  auto [result_bitmask, null_count] =
+    cudf::detail::valid_if(null_mask.begin(), null_mask.end(), thrust::identity<bool>{});
 
   return std::make_unique<cudf::column>(
     cudf::data_type{cudf::type_to_id<T>()},
     num_rows,
-    rmm::device_buffer(data.data(), num_rows * sizeof(stored_Type), rmm::cuda_stream_default),
-    std::move(result_bitmask));
+    data.release(),
+    profile.get_null_frequency() < 0 ? rmm::device_buffer{} : std::move(result_bitmask));
 }
 
-/**
- * @brief Class that holds string column data in host memory.
- */
-struct string_column_data {
-  std::vector<char> chars;
-  std::vector<cudf::size_type> offsets;
-  std::vector<cudf::bitmask_type> null_mask;
-  explicit string_column_data(cudf::size_type rows, cudf::size_type size)
+struct valid_or_zero {
+  template <typename T>
+  __device__ T operator()(thrust::tuple<T, bool> len_valid) const
   {
-    offsets.reserve(rows + 1);
-    offsets.push_back(0);
-    chars.reserve(size);
-    null_mask.insert(null_mask.end(), null_mask_size(rows), ~0);
+    return thrust::get<1>(len_valid) ? thrust::get<0>(len_valid) : T{0};
   }
 };
 
-/**
- * @brief Copy a string from one host-side "column" to another.
- *
- * Assumes that the destination null mask is initialized with all bits valid.
- */
-void copy_string(cudf::size_type src_idx,
-                 string_column_data const& src,
-                 cudf::size_type dst_idx,
-                 string_column_data& dst)
-{
-  if (!cudf::bit_is_set(src.null_mask.data(), src_idx))
-    cudf::clear_bit_unsafe(dst.null_mask.data(), dst_idx);
-  auto const str_len = src.offsets[src_idx + 1] - src.offsets[src_idx];
-  dst.chars.resize(dst.chars.size() + str_len);
-  if (cudf::bit_is_set(src.null_mask.data(), src_idx)) {
-    std::copy_n(
-      src.chars.begin() + src.offsets[src_idx], str_len, dst.chars.begin() + dst.offsets.back());
+struct string_generator {
+  char* chars;
+  thrust::minstd_rand engine;
+  thrust::uniform_int_distribution<unsigned char> char_dist;
+  string_generator(char* c, thrust::minstd_rand& engine)
+    : chars(c), engine(engine), char_dist(32, 137)
```

Comment:
Since this is more than just the alphanumeric characters it would be helpful to document the range of chars that could be generated.

Reply:
updated.